cmd/worker/pubsubextender/extender.go

```diff
@@ -41,15 +42,17 @@ func New(ctx context.Context, subURL string, sub *pubsub.Subscription) (*Extende
 		return nil, err
 	}
 
-	var d driver
-	switch u.Scheme {
-	case gcppubsub.Scheme:
-		d, err = newGCPDriver(u, sub)
-		if err != nil {
-			return nil, err
+	var d driver = &noopDriver{}
+	if featureflags.PubSubExtender.Enabled() {
```

Comment:
nit: can we follow the pattern of other usages of the feature flags? It's more like 
```
if !Enabled() { 
    return
}`

This also reduces the indentation

Reply:
Done. I moved the logic into its own function